src/qibolab/designs/qmrs.py

```diff
@@ -0,0 +1,160 @@
+from qibo.config import log, raise_error
+
+from qibolab.designs.abstract import AbstractInstrumentDesign
+from qibolab.platforms.utils import Channel
+
+
+class QMRSDesign(AbstractInstrumentDesign):
+    """Instrument design for Quantum Machines (QM) OPXs and Rohde Schwarz local oscillators.
+
+    IPs and other instrument related parameters are hardcoded in ``__init__`` and ``setup``.
+
+    Args:
+        address (str): Address and port for the QM OPX cluster.
+            Default is the current address for the instruments in TII lab.
+        simulation_duration (int): Duration for the simulation in ns.
+            If given the compiler simulator will be used instead of the actual hardware.
+            Default is ``None`` which falls back to the hardware.
+        cloud (bool): See :class:`qibolab.instruments.qmsim.QMSim` for details.
+            Relevant only when ``simulation_duration`` is given.
+
+    Attributes:
+        is_connected (bool): Boolean that shows whether instruments are connected.
+        opx (:class:`qibolab.instruments.qm.QMOPX`): Object used for controlling the QM OPXs.
+        local_oscillators (list): List of local oscillator objects.
+            instrument objects.
+    """
+
+    def __init__(self, address="192.168.0.1:80", simulation_duration=None, cloud=False):
+        from qibolab.instruments.rohde_schwarz import SGS100A
+
+        self.is_connected = False
+
+        # Instantiate QM OPX instruments
+        if simulation_duration is None:
+            from qibolab.instruments.qm import QMOPX
+
+            self.opx = QMOPX("qmopx", address)
+        else:
+            from qibolab.instruments.qmsim import QMSim
```

Comment:
> Is there a reason why you want to import only if you are using them?

The idea is that it allows using some of the functionality without having to install all the dependencies. For example, if the user does not have access to the lab, he may not install the dependencies related to connecting to instruments (such as the py-visa for LOs). By doing the import inside `if` he will still be able to run on simulation mode (which does not use the LOs).

If you believe this is a fair point, sure I will add a comment.

> However, Pylint should have warned you about this hint as well, are you not using it?

There is a pylint check in the CI, but as you can see it is passing, I am assuming because the warnings are disabled or not causing a fail. Fixing pylint warnings would for sure improve our code but I would guess it will be a lot of work at this point.

Reply:
A bit late, but I would also like to add that a conditional loading of libraries should help reduce the time required to load the application. We have had complaints in the past about the long time required to run a simple circuit and a significant part of it was the overhead required to load all libraries.